ag/validation.go

```diff
@@ -85,6 +82,23 @@ func Interceptor(logger *zap.Logger, userMap map[string]uint64) grpc.UnaryServer
 	}
 }
 
+// Returns modified metadata containing a valid user. Returns an error if the user is not authenticated.
+func UserValidation(meta metadata.MD, userMap map[string]uint64) (metadata.MD, error) {
+	token := meta.Get(Cookie)
+
+	if len(token) > 0 {
+		token := token[0]
+		user := userMap[token]
+		if user == 0 {
+			return nil, errors.New("Could not associate token with a user")
+		}
+		meta.Set(UserKey, strconv.FormatUint(user, 10))
+	} else {
```

Comment:
Seeing the logic now, you can write this code more clearly like this, avoiding else and indentations (this is recommended Go-style):

```golang
if len(token) == 0 {
 	return nil, errors.New("Request does not contain a session token")
}
token := token[0]
user := userMap[token]
if user == 0 {
 	return nil, errors.New("Could not associate token with a user")
}
meta.Set(UserKey, strconv.FormatUint(user, 10))
return meta, nil
```